examples/nuxt-openai/server/api/chat.ts

```diff
@@ -1,43 +1,33 @@
-// ./api/chat.ts
-import { Configuration, OpenAIApi } from 'openai-edge'
-import { OpenAIStream, streamToResponse } from 'ai'
+import {
+  type ChatCompletionRequestMessage,
+  Configuration,
+  OpenAIApi
+} from 'openai-edge'
+import { OpenAIStream } from 'ai'
 
-// Create an OpenAI API client (that's edge friendly!)
-const config = new Configuration({
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  apiKey: useRuntimeConfig().openaiApiKey
-})
-const openai = new OpenAIApi(config)
+let openai: OpenAIApi
 
-export default defineEventHandler(async (event: any) => {
+export default defineEventHandler(async event => {
+  if (!openai) {
+    const apiKey = useRuntimeConfig(event).openaiApiKey
```

Comment:
Would be nice to use a lazy event handler instead of global caching. See this https://github.com/pi0/nuxt-openai-demo/blob/main/server/api/chat.ts#L8

Reply:
It does work with Cloudflare (module and workers) as well (not passing event within lazy wrapper is acceptable since env is already available.